sklearn/pipeline.py

```diff
@@ -220,20 +236,44 @@ def _final_estimator(self):
 
     def _fit(self, X, y=None, **fit_params):
         self._validate_steps()
+        # Setup the memory
+        memory = self.memory
+        if memory is None:
+            memory = Memory(cachedir=None, verbose=0)
+        elif isinstance(memory, six.string_types):
+            memory = Memory(cachedir=memory, verbose=0)
+        elif not isinstance(memory, Memory):
+            raise ValueError('memory is either `str` or a `joblib.Memory`'
+                             ' instance, got: %s' % memory)
+        # Decorate _fit_transform_one to be able to use it
+        # with cache
+        fit_transform_one = memory.cache(_fit_transform_one)
```

Comment:
`_fit_transform_one` has a `name` parameter that is not used. This means that this will result in cache misses if you change the name of the step. Maybe we should fix this in this PR? You could do the same for `_transform_one` while you are at it.